.eslintrc

```diff
@@ -14,20 +14,347 @@
+		"arrow-parens": [
+			"warn",
+			"as-needed",
+			{
+				"requireForBlockBody": true
+			}
+		],
+		"arrow-spacing": [
+			"error",
+			{
+				"after": true,
+				"before": true
+			}
+		],
+		"block-scoped-var": "off",
+		"block-spacing": [
+			"error",
+			"always"
+		],
+		"brace-style": [
+			"warn",
+			"stroustrup"
+		],
+		"callback-return": "off",
+		"camelcase": "error",
+		"capitalized-comments": "off",
+		"class-methods-use-this": "error",
+		"comma-dangle": "off",
+		"comma-spacing": [
+			"warn",
+			{
+				"after": true,
+				"before": false
+			}
+		],
+		"comma-style": [
+			"error",
+			"last"
+		],
+		"complexity": "off",
+		"computed-property-spacing": [
+			"error",
+			"never"
+		],
+		"consistent-return": "error",
+		"consistent-this": "error",
+		"curly": "off",
+		"default-case": "off",
+		"dot-location": [
+			"error",
+			"property"
+		],
+		"dot-notation": "error",
+		"eol-last": "error",
+		"eqeqeq": "off",
+		"func-call-spacing": "error",
+		"func-name-matching": "error",
+		"func-names": [
+			"error",
+			"never"
+		],
+		"func-style": "off",
+		"function-paren-newline": "off",
+		"generator-star-spacing": [
+			"error",
+			{
+				"before": false,
+				"after": true
+			}
+		],
+		"global-require": "off",
+		"guard-for-in": "off",
+		"handle-callback-err": "error",
+		"id-blacklist": "error",
+		"id-length": "off",
+		"id-match": "error",
+		"implicit-arrow-linebreak": [
+			"error",
+			"beside"
+		],
+		"indent": [
+			"error",
+			"tab"
+		],
+		"init-declarations": "off",
+		"jsx-quotes": "error",
+		"key-spacing": "warn",
+		"keyword-spacing": "warn",
+		"line-comment-position": "off",
+		"linebreak-style": [
+			"error",
+			"unix"
+		],
+		"lines-around-comment": "error",
+		"lines-around-directive": "error",
+		"lines-between-class-members": "error",
+		"max-depth": "off",
+		"max-len": "off",
+		"max-lines": "off",
+		"max-nested-callbacks": "error",
+		"max-params": "off",
+		"max-statements": "off",
+		"max-statements-per-line": "off",
+		"multiline-comment-style": "off",
+		"multiline-ternary": [
+			"error",
+			"always-multiline"
+		],
+		"new-parens": "off",
+		"newline-per-chained-call": [
+			"error",
+			{
+				"ignoreChainWithDepth": 3
+			}
+		],
+		"no-alert": "off",
+		"no-array-constructor": "error",
+		"no-await-in-loop": "warn",
+		"no-bitwise": "off",
+		"no-buffer-constructor": "error",
+		"no-caller": "error",
+		"no-confusing-arrow": "error",
+		"no-constant-condition": [
+			"error",
+			{
+				"checkLoops": false
+			}
+		],
+		"no-control-regex": "off",
+		"no-continue": "off",
+		"no-div-regex": "off",
+		"no-duplicate-imports": "error",
+		"no-else-return": "off",
+		"no-empty": "off",
+		"no-empty-function": "off",
+		"no-eq-null": "error",
+		"no-eval": "error",
+		"no-extend-native": "error",
+		"no-extra-bind": "error",
+		"no-extra-label": "error",
+		"no-extra-parens": "off",
+		"no-floating-decimal": "error",
+		"no-implicit-globals": "error",
```

Comment:
Any top-level var or function (such as `doWeb`) will trigger this rule.

Reply:
Maybe we should default to sourceType `module` rather than `script`

Reply:
I've removed `no-implicit-globals`.